Request to combine code refactor and party system

MyChakoTaco opened the pull request:
Refactor to use bot.command instead of checking every message and seeing what the prefix is

Added a party system that can be called by anyone and joined/left by reaction emojis

MyChakoTaco committed:
Refactor code to use bot.command

- Major refactor to format code to match that of discord.py's bot.command structure
- Remove try/catch for common branches and instead anticipate incorrect argument input
- Remove regex as it's no longer needed
- Format 'games' command to have string padding so that number of users per game line up nicely

MyChakoTaco committed:
Add Party System

- Add party.py object for handling state and embed display
- Update roles.py to include more fields so that it can be handled by preset
- Add party command, on_reaction_add event and on_reaction_remove event for party commands
- Add background loop to clean-up inactive parties
- Clean up code refactor a bit more

MyChakoTaco committed:
Merge branch 'master' into master

danny-tom commented on .gitignore:
```diff
@@ -1 +1,4 @@
 __pycache__/
+.env
+.vscode/launch.json
+.env
```
Duplicate .env

MyChakoTaco replied:
Will fix

danny-tom commented on roles.py:
```diff
@@ -6,12 +6,33 @@
-              OVERWATCH,
-              PROJECTWINTER
-              ]
+    VALORANT,
```
Are you using a linter? I've been using Flake8 if we want to standardize these indents and other stuff.

MyChakoTaco replied:
I'm just using microsoft visual code, but i've been eyeballing it. i can probably find the Flake8 thing and use it.

danny-tom commented on party.py:
```diff
@@ -0,0 +1,102 @@
+    @staticmethod
+    def __getPreset(name):
+        for preset in roles.ROLES_LIST:
+            if name.lower() == preset.name.lower():
```
casefold for safety? I dunno. https://stackoverflow.com/questions/45745661/python-lower-vs-casefold-in-string-matching-and-converting-to-lowercase

MyChakoTaco replied:
casefold sounds good. i'm not sure if roles can have emoji's, but let's just opt for safer

danny-tom commented on charon.py:
```diff
@@ -221,4 +185,44 @@ async def on_member_join(member):
+BACKGROUND_LOOP_TIME = 60
+# This function checks if parties are inactive
+# and cleans up them up if they are
+async def update_parties():
```
There is a decorator that might be available here
https://discordpy.readthedocs.io/en/latest/ext/tasks/index.html?highlight=task#discord.ext.tasks.loop

MyChakoTaco replied:
will check it out

danny-tom reviewed (changes requested):
Mostly looking to standardize some spacing. Some other nitpicks

MyChakoTaco committed:
Change Request

- Remove duplicate .env in gitignore
- Format code on flake8 linter
- Use casefold instead of lower
- Use loop decorator instead of asynchio

MyChakoTaco committed:
Merge branch 'master' of https://github.com/MyChakoTaco/charon

MyChakoTaco commented:
Made the requested changes

danny-tom reviewed (approved):
Thanks for making the changes. LGTM
